relay-server/src/middlewares/body_timing.rs

```diff
@@ -0,0 +1,288 @@
+use crate::statsd::RelayTimers;
+use axum::body::{Body, HttpBody};
+use axum::http::Request;
+use hyper::body::Frame;
+use relay_statsd::metric;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+use std::time::Duration;
+use tokio::time::Instant;
+use tower::{Layer, Service};
+
+/// Middleware layer that wraps the request [`Body`] to measure body reading duration.
+///
+/// Metrics are tagged with a completion status to track whether the body:
+/// - was read to completion ("completed")
+/// - failed during reading ("failed")
+/// - was dropped before completion ("dropped")
```

Comment:
Would be good if you also mentioned which tags there are if you already mention the status.